crates/services/gas_price_service/src/v1/da_source_service/service.rs

```diff
@@ -124,12 +128,9 @@ where
         &self,
         da_block_costs: Vec<DaBlockCosts>,
     ) -> Result<impl Iterator<Item = DaBlockCosts>> {
-        let latest_l2_height = *self
-            .latest_l2_height
-            .lock()
-            .map_err(|err| anyhow::anyhow!("lock error: {:?}", err))?;
+        let latest_l2_height = self.latest_l2_height.load(Ordering::SeqCst);
```

Comment:
```suggestion
        let latest_l2_height = self.latest_l2_height.load(Ordering::Acquire);
```

SeqCst is not needed unless you have at least two atomic variables. The purpose of SeqCst is to make sure that different threads do not see updates to different variables in different order. See https://stackoverflow.com/questions/14861822/acquire-release-versus-sequentially-consistent-memory-order for an example. 

(Just mentioning this, overall I think we can keep SeqCst)

Reply:
Updated in 5eeba9e155c1b75d1f0ff2031a41582a9179c995